doc/advanced.rst

```diff
@@ -383,33 +383,46 @@ Tags
 ----
 
 One of the most exciting features of a template engine like Twig is the
-possibility to define new language constructs. This is also the most complex
+possibility to define new **language constructs**. This is also the most complex
 feature as you need to understand how Twig's internals work.
 
-.. tip::
+Most of the time though, a tag is not needed:
+
+* If your tag generate some output, use a **function** instead.
+
+* If your tag modifies some content and outputs it, use a **filter** instead.
```

Comment:
a filter must not output the modified content. It must return it. Otherwise, it breaks both chaining and autoescaping (which is related as autoescaping is actually implemented by changing filters)

Reply:
`outputs` => `returns`